README.md

```diff
@@ -1,48 +1,70 @@
-
 ## System Requirements
 
-The project uses poetry for Python to create an isolated environment and manage package dependencies. To prepare your system, ensure you have an official distribution of Python version 3.7+ and install Poetry using one of the following commands (as instructed by the [poetry documentation](https://python-poetry.org/docs/#system-requirements)):
+You need python 3.7+, poetry, and some local configuration.  If you are using `bash` and have a working `make` then just issue
 
-### Poetry installation (Bash)
+    $ make
+    
+... and it will attempt most of the setup and initialization for you.  On your first run it will fail to authenticate against trello and you then need to provide your credentials in the `.env` file.
 
-```bash
-curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python -
-```
+If you don't have `bash` and `make`, or if this didn't work, then the manual steps are listed below.
 
-### Poetry installation (PowerShell)
+## Trello Setup
 
-```powershell
-(Invoke-WebRequest -Uri https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py -UseBasicParsing).Content | python -
-```
+If using the trello persistence layer, you need to configure trello in a particular way:
+
+1. Create a board (any name) with at least one list called "Not Started".
+2. Create an integration via the powerups pages at https://trello.com/power-ups/admin that can access that board
+3. Note your API Key and your Token as you will need them for _Credentials_ below (follow the hyperlink next to the API key in the powerup admin pages)
+
+## Credentials
+
+To run the app you need a `.env` file containing at least `FLASK_APP`, `FLASK_ENV`, `SECRET_KEY`, `TRELLO_API_KEY`, and `TRELLO_TOKEN`.  The `Makefile` will create this if it doesn't exist. 
+
+The `FLASK`* values are fine to leave as they are, but the `TRELLO_` values need to be initialized with your API key and Token from _Trello Setup_ above.  Just paste the keys after the equals signs, no quotes.
 
-## Dependencies
+There's also a [SECRET_KEY](https://flask.palletsprojects.com/en/1.1.x/config/#SECRET_KEY) variable which is used to encrypt the flask session cookie.
 
-The project uses a virtual environment to isolate package dependencies. To create the virtual environment and install required packages, run the following from your preferred shell:
+## Find the board's ID
+
+Once you have the `TRELLO_API_KEY` and `TRELLO_TOKEN` set up, you can try
 
 ```bash
-$ poetry install
+$ make choose-board
 ```
 
-You'll also need to clone a new `.env` file from the `.env.template` to store local configuration options. This is a one-time operation on first setup:
+This lists the boards that you were authorised to see, and asks you to choose one, then writes the config to the app. You see something like the following:
```

Comment:
This feels like it's just an alternative hand-rolled approach to setting these as environment variables - I like the idea of having a helpful script to find the values, but I think putting them in the `.env` file rather than into a Python file would be simpler. For example, at some point we'll deploy this to different environments; we now need a custom approach at each stage for generating the relevant Python file without a human running this script whereas most tools have easy existing solutions for setting environment variables. I'd also worry that it will encourage us to bake these variables into some versions of the code, which could make swapping them generally harder

Reply:
Yeah. I hadn't heard of .env at the time I hand rolled it.  I'm a bit socially clueless and my coding is 15 years out of date (though I did get off python 2.7 in the end) so not good at telling whether .env is standard or some additional wonk.  For example, I'm in python here, why use something else that I have to pip install, why not write some python :-) but we can call it house style and I'll happily comply

Reply:
hmm, taking a look at it, the output of trello-init includes a dict from trello list name to ID, so not sure how I would encode that in .env.  Also it needs .env to be partially initialized so that trelloinit can connect to the board.  Maybe it would be more palatable/conventional as a config.json file?

Reply:
raised as an issue for later